blockchain/tx_pool.go

```diff
@@ -924,10 +933,114 @@ func (pool *TxPool) HandleTxMsg(txs types.Transactions) {
 	if pool.config.DenyRemoteTx {
 		return
 	}
+
+	// Filter spam txs based on to-address of failed txs
+	spamThrottler := GetSpamThrottler()
+	if spamThrottler != nil {
+		pool.mu.RLock()
+		poolSize := uint64(len(pool.all))
+		pool.mu.RUnlock()
+
+		// Activate spam throttler when pool has enough txs
+		if poolSize > uint64(spamThrottler.config.ActivateTxPoolSize) {
+			allowTxs, throttleTxs := spamThrottler.classifyTxs(txs)
+
+			for _, tx := range throttleTxs {
+				select {
+				case spamThrottler.throttleCh <- tx:
+				default:
+					logger.Trace("drop a tx when throttleTxs channel is full", "txHash", tx.Hash())
+				}
+			}
+
+			txs = allowTxs
+		}
+	}
+
+	// TODO-Klaytn: Consider removing the next line and move the above logic to `addTx` or `AddRemotes`
```

Comment:
I think moving this line is not good. Because it is better to recover as soon as possible

Reply:
I think you want to consider the throttled transactions. For that reason, we can consider it later.

Reply:
I added this comment because I think calling `sendCacher.recover` here doesn't make processing faster but enqueues senderCacher task channels twice.
`sendCacher.recover` is called again after checking txpool capacity in `pool.addTxs`. And, `pool.txMsgCh <- txs` is hard to block something because the channel pops txs as soon as the channel is filled without any blocker.